sycl/test/basic_tests/diagnostics/device-check.cpp

```diff
@@ -0,0 +1,38 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: env SYCL_DEVICE_TYPE=cpu %t.out
+// RUN: env SYCL_DEVICE_TYPE=gpu %t.out
+// RUN: env SYCL_DEVICE_TYPE=acc %t.out
+// RUN: env SYCL_DEVICE_TYPE=host %t.out
+// RUN: env SYCL_DEVICE_TYPE=CPU %t.out
+// RUN: env SYCL_DEVICE_TYPE=GPU %t.out
+// RUN: env SYCL_DEVICE_TYPE=ACC %t.out
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out
+// RUN: env SYCL_DEVICE_TYPE=Cpu %t.out
+// RUN: env SYCL_DEVICE_TYPE=Gpu %t.out
+// RUN: env SYCL_DEVICE_TYPE=Acc %t.out
+// RUN: env SYCL_DEVICE_TYPE=Host %t.out
+
```

Comment:
I believe the separate negative test can be replaced with an addition like:
```suggestion
// RUN: not %clangxx -fsycl %s -o %t.out 2>&1 | FileCheck %s
// CHECK: SYCL_DEVICE_TYPE is not available.  Must be GPU, CPU, ACC or HOST
```

Reply:
Hi Artem,

Thanks for the code review comments. 

Personally, I prefer several smaller tests to one larger test.  I particular, I separate negative tests from positive tests.  Is there a specific "rule of thumb" about this in LLVM?  If Icould keep two test cases, I would prefer that. 
I appreciate your suggestion on using FileCheck.  That had not occurred to me.

Reply:
@glyons-intel, feel free to treat my feedback as optional, for I have no strong objections to the current patch. https://llvm.org/docs/TestingGuide.html#writing-new-regression-tests was the inspiration for this particular comment, but again, "rule of thumb" isn't really implied there.

Reply:
This suggestion makes sense to me.
1. It removes duplication - device-check.cpp and device-type.cpp source code is the same.
2. It saves on compilation - one of the goals of testing system is to be as fast as possible, so saving compilation is good thing to do.